sys/net/gnrc/network_layer/ipv6/nib/nib.c

```diff
@@ -1621,7 +1621,11 @@ static uint32_t _handle_pio(gnrc_netif_t *netif, const icmpv6_hdr_t *icmpv6,
     valid_ltime = byteorder_ntohl(pio->valid_ltime);
     pref_ltime = byteorder_ntohl(pio->pref_ltime);
     if ((pio->len != NDP_OPT_PI_LEN) || (icmpv6->type != ICMPV6_RTR_ADV) ||
-        ipv6_addr_is_link_local(&pio->prefix) || (valid_ltime < pref_ltime)) {
+        ipv6_addr_is_link_local(&pio->prefix) || (valid_ltime < pref_ltime)
+        || (gnrc_netif_is_6ln(netif) &&
+        /*https://datatracker.ietf.org/doc/html/rfc6775#section-5.4*/
+        (pio->flags & NDP_OPT_PI_FLAGS_L))
+        ) {
```

Comment:
```suggestion
        ipv6_addr_is_link_local(&pio->prefix) || (valid_ltime < pref_ltime) ||
        /* https://datatracker.ietf.org/doc/html/rfc6775#section-5.4 */
        (gnrc_netif_is_6ln(netif) && (pio->flags & NDP_OPT_PI_FLAGS_L)) {
```

makes it a bit more readable.
Those monster if-statements should probably be broken up into smaller ones, but they are littered through the network code already.